### ISSUE TITLE
No parameter for interlaced HEVC encoding on ffmpeg plugin 

### ISSUE THREAD

**Issue by edisongjs** [intel/SVT-HEVC#137]
Hi,

I do not see a parameter to make an interlaced HEVC encoding (1080i29.97). For example, I could generate interlaced HEVC encoding with x265 with below command. What would be a command for SVT? SVT sample application has a parameter ("-interlaced-video") to signal interlaced video though.

Parameter for x265:
-x265-params interlace=1

SVT FFMPEG help message (built on top of ffmpeg 4.1)
==================
# /tools/ffmpeg_4.1_svt-hevc -h encoder=libsvt_hevc
ffmpeg version n4.1-64-gbcfd82b Copyright (c) 2000-2019 the FFmpeg developers
  built with gcc 5.4.0 (Ubuntu 5.4.0-6ubuntu1~16.04.10) 20160609
  configuration: --prefix=/root/ffmpeg_build --extra-cflags=-I/root/ffmpeg_build/include --pkg-config='pkg-config --static' --extra-ldflags=-L/root/ffmpeg_build/lib --bindir=/root/bin --enable-gpl --enable-nonfree --enable-libsvthevc
  libavutil      56. 22.100 / 56. 22.100
  libavcodec     58. 35.100 / 58. 35.100
  libavformat    58. 20.100 / 58. 20.100
  libavdevice    58.  5.100 / 58.  5.100
  libavfilter     7. 40.101 /  7. 40.101
  libswscale      5.  3.100 /  5.  3.100
  libswresample   3.  3.100 /  3.  3.100
  libpostproc    55.  3.100 / 55.  3.100
Encoder libsvt_hevc [SVT-HEVC(Scalable Video Technology for HEVC) encoder]:
    General capabilities: delay threads
    Threading capabilities: auto
    Supported pixel formats: yuv420p yuv420p10le
libsvt_hevc AVOptions:
  -vui               <boolean>    E..V..... Enable vui info (default true)
  -aud               <boolean>    E..V..... Include AUD (default false)
  -hielevel          <int>        E..V..... Hierarchical prediction levels setting (from 1 to 4) (default 4level)
     flat                         E..V.....
     2level                       E..V.....
     3level                       E..V.....
     4level                       E..V.....
  -la_depth          <int>        E..V..... Look ahead distance [0, 256] (from -1 to 256) (default -1)
  -preset            <int>        E..V..... Encoding preset [0, 12] (e,g, for subjective quality tuning mode and >=4k resolution), [0, 10] (for >= 1080p resolution), [0, 9] (for all resolution and modes) (from 0 to 12) (default 9)
  -profile           <int>        E..V..... Profile setting, Main Still Picture Profile not supported (from 1 to 2) (default main10)
     main                         E..V.....
     main10                       E..V.....
  -tier              <int>        E..V..... Set tier (general_tier_flag) (from 0 to 1) (default main)
     main                         E..V.....
     high                         E..V.....
  -level             <int>        E..V..... Set level (level_idc) (from 0 to 255) (default 0)
     1                            E..V.....
     2                            E..V.....
     2.1                          E..V.....
     3                            E..V.....
     3.1                          E..V.....
     4                            E..V.....
     4.1                          E..V.....
     5                            E..V.....
     5.1                          E..V.....
     5.2                          E..V.....
     6                            E..V.....
     6.1                          E..V.....
     6.2                          E..V.....
  -rc                <int>        E..V..... Bit rate control mode (from 0 to 1) (default cqp)
     cqp                          E..V.....
     vbr                          E..V.....
  -qp                <int>        E..V..... QP value for intra frames (from 0 to 51) (default 32)
  -sc_detection      <boolean>    E..V..... Scene change detection (default false)
  -tune              <int>        E..V..... Quality tuning mode (from 0 to 1) (default objective)
     subjective                   E..V..... Subjective quality mode
     objective                    E..V..... Objective quality mode for PSNR / SSIM / VMAF benchmarking
  -bl_mode           <boolean>    E..V..... Random Access Prediction Structure type setting (default false)
  -forced-idr        <boolean>    E..V..... If forcing keyframes, force them as IDR frames. (default false)

**Comment by tianjunwork** [comment]
@jsunintel Please help with the request here. Thanks.

**Comment by jsunintel** [comment]
OK

**Comment by tianjunwork** [comment]
@jsunintel Yes, you are correct. The logic of formatting of input data is at application level, not svt lib. Do you think the ffmpeg plugin can process interlaced data?

**Comment by jsunintel** [comment]
> 
> 
> @jsunintel Yes, you are correct. The logic of formatting of input data is at application level, not svt lib. Do you think the ffmpeg plugin can process interlaced data?

Let me have a check.

**Comment by ValZapod** [comment]
I doubt HEVC interlacing is supported in anything besides https://sourceforge.net/p/mjpeg/mailman/message/9927377/ and LG, Panasonic TVs.

**Comment by xsistema** [comment]
Music Box Russia channel over satellite transmits in HEVC 1920x1080 25fps interlaced - and after recording VLC recognizes file as 50 fps, and resolution 1920x540 - half a height. But on satellite tuner the player works ok - it plays the file as 1920x1080 25fps... When we can expect support for HEVC/H265 interlaced?

**Comment by tianjunwork** [comment]
Hi @xsistema, could you test below command line `SvtHevcEncApp -i interlaced.yuv -w 1920 -h 540 -interlaced-video 1 -b out.bin` to see if it works? If so, we will work on exposing the `-interlaced-video` interface through ffmpeg.
